core/common/src/main/scala/org/typelevel/otel4s/Attribute.scala

```diff
@@ -28,3 +28,45 @@ package org.typelevel.otel4s
   *   the type of the attribute's value. One of [[AttributeType]]
   */
 final case class Attribute[A](key: AttributeKey[A], value: A)
+
+object Attribute {
+
+  @annotation.implicitNotFound("""
+Could not find the `KeySelect` for ${A}. The `KeySelect` is defined for the following types:
+String, Boolean, Long, Double, List[String], List[Boolean], List[Long], List[Double].
+""")
+  sealed trait KeySelect[A] {
```

Comment:
Well, it's not select. But I didn't come up with a decent name.

Reply:
I don't think it's a huge deal, because it shouldn't be referenced often in end user code?